spec/jobs/education_form/create_spool_submissions_report_spec.rb

```diff
@@ -35,40 +36,57 @@
         )
       end
 
-      describe '#perform' do
-        before do
-          expect(FeatureFlipper).to receive(:send_edu_report_email?).once.and_return(true)
-        end
-        after do
-          File.delete(filename)
-        end
+      it 'should check for stem submissions' do
+        data = subject.create_csv_array
+        expect(data[:stem_exists]).to eq(false)
+      end
 
-        let(:filename) { "tmp/spool_reports/#{time.to_date}.csv" }
+      it 'should recognize 1995s as STEM submission' do
+        create(:education_benefits_claim_1995s, processed_at: time.beginning_of_day)
+        data = subject.create_csv_array
+        expect(data[:stem_exists]).to eq(true)
+      end
 
-        def perform
-          stub_reports_s3(filename) do
-            subject.perform
-          end
-        end
+      it 'should recognize 1995 with STEM data as STEM submission' do
+        create(:education_benefits_claim_1995stem, processed_at: time.beginning_of_day)
+        data = subject.create_csv_array
+        expect(data[:stem_exists]).to eq(true)
+      end
+    end
 
-        it 'should create a csv file' do
-          perform
-          data = subject.create_csv_array
-          csv_array = data[:csv_array]
-          csv_string = CSV.generate do |csv|
-            csv_array.each do |row|
-              csv << row
-            end
-          end
+    describe '#perform' do
+      let(:filename) { "tmp/spool_reports/#{time.to_date}.csv" }
+
+      before do
+        expect(FeatureFlipper).to receive(:send_edu_report_email?).once.and_return(true)
+      end
+
+      after do
+        File.delete(filename)
```

Comment:
You're deleting here, but the create is in a `let` -- wouldn't it make just as much sense to put it in the `before`?

Do you need this file for every example in this describe block? If not it would be a lot clearer to include it only in the example(s) that need it.

Al `before(:all)` would let you create it once, followed by an `after(:all)` to clean it up, all inside this `describe` block. That can get hairy when mixed with (:each) but in this case you're saving a lot of file creation steps too.

Reply:
Both examples create a spool file, so deleting it in the `after` makes sense. The `let` here is only the name of the file.

Reply:
@johnpaulashenfelter have you had a chance to look at this?  Thanks!

Reply:
👍